test/e2e/upgrade_test.go

```diff
@@ -150,3 +168,34 @@ func (tc *testContext) scaleWMCODeployment(desiredReplicas int32) error {
 
 	return err
 }
+
+// deployWindowsWorkloadTester tests if the Windows Webserver deployment is available.
+// This is achieved by creating a Job object that continuously pings the webserver every 5 seconds.
+func (tc *testContext) deployWindowsWorkloadTester() error {
+	// Get Windows Webserver deployment
+	deployment, err := tc.kubeclient.AppsV1().Deployments(tc.workloadNamespace).Get(context.TODO(), "win-webserver-deployment",
+		metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "error getting Windows Webserver deployment")
+	}
+
+	// Create a clusterIP service which can be used to reach the Windows webserver
+	intermediarySVC, err := tc.createService(deployment.Name, v1.ServiceTypeClusterIP, *deployment.Spec.Selector)
+	if err != nil {
+		return errors.Wrap(err, "could not create service ")
+	}
+
+	// Create a Job to ping Windows webserver
+	// Ping Windows webserver every 5 seconds. If the webserver is not accessible for few minutes
+	// exit the pod with an error.
+	command := []string{"bash", "-c",
+		" while true; do curl " + intermediarySVC.Spec.ClusterIP + "; if [ $? != 0 ]; then sleep 60; curl " +
+			intermediarySVC.Spec.ClusterIP + "|| exit 1;" + " fi; sleep 5; done"}
```

Comment:
`If the webserver is not accessible for few minutes`
I'm not sure if this comment is accurate to the functionality

It doesnt seem to be keeping track of how long the service is not accessible, but instead checking every minute to see if its down. It could be down for a second at the wrong time and cause a failure. 

I almost feel like you should try using smaller intervals and keep track of how many intervals the service was down. Once it hits a certain number of misses, exit 1

Reply:
The ping happens every 5 seconds.

If the curl command `errors` out, only then we wait for 1 min.
Also note that the `connect-timeout` for curl is around 2 mins. So if it is down for a second at a wrong time, it should be handled by the subsequent `curl` call.

Reply:
You're right, not an issue then